GitUI/CommandsDialogs/RevisionDiffControl.cs

```diff
@@ -449,8 +453,21 @@ private void ResetSelectedItemsTo(bool actsAsChild)
             RequestRefresh();
         }
 
+        private Task BlameSelectedFileDiffAsync()
+        {
+            BlameControl.Visible = true;
+            DiffText.Visible = false;
+            GitRevision rev = DiffFiles.SelectedItem.SecondRevision.IsArtificial
+                ? _revisionGrid.GetActualRevision(_revisionGrid.CurrentCheckout)
+                : DiffFiles.SelectedItem.SecondRevision;
+            BlameControl.LoadBlame(rev, null, DiffFiles.SelectedItem.Item.Name, _revisionGrid, null, DiffText.Encoding);
+            return Task.CompletedTask;
+        }
```

Comment:
This is really confusing - there is nothing async in this method, and everything must be done on the UI thread.
Please change it into a sync method, and update the the call site to invoke it async, if necessary.

Reply:
This was done in the same way as `ShowSelectedFileDiffAsync()`, the call in `blameToolStripMenuItem_Click()` need to async anyway (even if this is sync).
Cancelation token should be added to LoadBlame too, similar to how `ViewChangesAsync()` was changed.
(that will require Git commands to be cancellable first to be really useful)

So I would prefer to keep it as is.

Reply:
Changed this - there are operations blocking the UI thread, but not for long.